wordpress-mobile/WordPress-Android
# Jetpack App (Basics): Updated internal "App" Icons

Fixes #14475 

This PR refactors the use of `ic_my_sites_white_24dp` so the WordPress and the Jetpack dimensions show the proper icons when needed.

**Implementation Details**
WordPress Main repo
- Created new vector asset `ic_wordpress_white_24dp` as a copy `ic_my_sites_white_24dp` 
- Created new vector asset `ic_app_white_24dp` as a copy of `ic_my_sites_white_24dp`
- For “My sites” locations
--`ic_plans_white_24dp` - no changes
-- View Admin Icon - used `ic_wordpress_white_24dp` directly
- Media_menu, Followers (stats)
-- Changed `ic_my_sites_white_24dp` to `ic_wordpress_white_24dp`
- About
-- Changed `ic_my_sites_white_24dp` to `ic_app_white_24dp`
- Notification Locations
-- Changed `ic_my_sites_white_24dp` to `ic_app_white_24dp`
- Navigation “My Sites” Tab
-- No changes - it uses `ic_my_sites_white_24dp`
- Deleted `ic_about_app_white_24dp`

Jetpack dimension
    - Created new vector asset `ic_app_white_24dp` as a copy of `ic_plans_white_24dp` (jetpack grid icon)
    - Created new vector asset `ic_my_sites_white_24dp` as a copy of `ic_plans_white_24dp` (jetpack grid icon)

**Screenshots**

WordPress | Jetpack
-----|-----
<img width="250" height="500" alt="wordpress_about" src="https://user-images.githubusercontent.com/506707/114939615-b5b1ef00-9e0e-11eb-9b1a-7d024cae618c.png"> | <img width="250" height="500" alt="jetpack_about" src="https://user-images.githubusercontent.com/506707/114939626-b9457600-9e0e-11eb-82de-f27c90ccc65f.png">
<img width="250" height="500" alt="wordpress_my_site" src="https://user-images.githubusercontent.com/506707/115459860-8972e580-a1f5-11eb-8688-34b4f3062687.png"> | <img width="250" height="500" alt="jetpack_my_site" src="https://user-images.githubusercontent.com/506707/115459844-84ae3180-a1f5-11eb-8a7b-44da646ee797.png">

<img width="250" height="500" alt="wordpress_view_admin" src="https://user-images.githubusercontent.com/506707/115460207-fc7c5c00-a1f5-11eb-8547-7686dd096841.png"> | <img width="250" height="500" alt="jetpack_view_admin" src="https://user-images.githubusercontent.com/506707/115460211-fdad8900-a1f5-11eb-8aef-f2672e7c5d87.png">

<img width="250" height="500" alt="wordpress_followers" src="https://user-images.githubusercontent.com/506707/115460370-30578180-a1f6-11eb-84c5-a0e4a3035094.png"> | <img width="250" height="500" alt="jetpack_view_followers" src="https://user-images.githubusercontent.com/506707/115460424-3fd6ca80-a1f6-11eb-9d37-4f1a815fcf6f.png">

<img width="250" height="500" alt="wordpress_media_menu" src="https://user-images.githubusercontent.com/506707/115460473-511fd700-a1f6-11eb-8a38-089ad7232d43.png"> | <img width="250" height="500" alt="jetpack_view_media-menu" src="https://user-images.githubusercontent.com/506707/115460478-52e99a80-a1f6-11eb-87e1-cff3494eb325.png">

*WordPress Only Screenshots*
Plans | Status Bar Notif | Notification
-----|-----|-----
<img width="250" height="500" alt="wordpress_plans" src="https://user-images.githubusercontent.com/506707/115460785-a9ef6f80-a1f6-11eb-83fe-bf957e243909.png"> | <img width="250" height="500" alt="wordpress_status" src="https://user-images.githubusercontent.com/506707/115460790-acea6000-a1f6-11eb-868d-28d55c1ec17d.png"> | <img width="250" height="500" alt="jetpack_notification" src="https://user-images.githubusercontent.com/506707/115460793-aeb42380-a1f6-11eb-9eda-e92f6786dde5.png">

**To test:**
Scenario 1: WordPress App
- Launch the App
- On the My Site View verify
-- The Plans icon shows the Jetpack icon
-- The View Admin icon shows the WordPress Icon
-- The My Site navigation Tab shows the WordPress Icon

- Navigate to me -> App Settings -> About The App -> Tap WordPress for Android and verify
-- The About icon shows the WordPress icon

- Navigate to back to My Site -> Stats and verify:
-- The Follower Totals block shows the WordPress Icon

- Navigate back to My Site and Tap on the Site Icon -> Site Icon dialog select "Change" (you may be prompted to accept permissions first) and verify:
-- The WordPress icon is shown in the toolbar

- From another device, perform an action that will trigger a push notification to the device you are testing on. Verify:
-- The push notification in the status bar shows the WordPress icon
-- The push notification preview shows the WordPress icon
-------------------------------

Scenario 2: Jetpack App
- Launch the App (you'll need to download this build and run locally until CI starts creating an APK)
- On the My Site View verify
-- The Plans icon shows the Jetpack icon
-- The View Admin icon shows the WordPress Icon
-- The My Site navigation Tab shows the Jetpack Icon

- Navigate to me -> App Settings -> About The App -> Tap Jetpack for Android and verify
-- The About icon shows the Jetpack icon

- Navigate to back to My Site -> Stats and verify:
-- The "Follower Totals" block shows the WordPress Icon

- Navigate back to My Site and Tap on the Site Icon -> Site Icon dialog select "Change" (you may be prompted to accept permissions first) and verify:
-- The WordPress icon is shown in the toolbar

- I was unable to test push notifications for Jetpack, so the icon verification test will need to be performed in the future.
 
## Regression Notes
1. Potential unintended areas of impact
- Areas in the app that need to use the App icon but were not referencing `ic_my_sites_white_24dp`, to they could not be identified.
- A notification type specifically requires a WordPress icon and not an "App" icon.

2. What I did to test those areas of impact (or what existing automated tests I relied on)
Manual testing

3. What automated tests I added (or what prevented me from doing so)
N/A

PR submission checklist:
